metatensor-torch/include/metatensor/torch/atomistic/model.hpp

```diff
@@ -0,0 +1,257 @@
+
+private:
+    double model_to_engine_ = 1.0;
+};
+
+
+/// Options requested by the simulation engine when running with a model
+class METATENSOR_TORCH_EXPORT ModelEvaluationOptionsHolder: public torch::CustomClassHolder {
+public:
+    ModelEvaluationOptionsHolder() = default;
+
+    /// Initialize `ModelEvaluationOptions` with the given data
+    ModelEvaluationOptionsHolder(
+        std::string length_unit,
+        torch::Dict<std::string, ModelOutput> outputs,
+        torch::optional<TorchLabels> selected_atoms
+    );
+
+    ~ModelEvaluationOptionsHolder() override = default;
+
+    /// unit of lengths the engine uses for the model input
+    std::string length_unit;
+
+    /// requested outputs for this run and corresponding settings
+    torch::Dict<std::string, ModelOutput> outputs;
+
+    /// Only run the calculation for a selected subset of atoms. If this is set
+    /// to `None`, run the calculation on all atoms. If this is a set of
+    /// `Labels`, it will have two dimensions named `"system"` and `"atom"`,
+    /// containing the 0-based indices of all the atoms in the selected subset.
+    torch::optional<TorchLabels> get_selected_atoms() const {
+        return selected_atoms_;
+    }
+
+    /// Setter for `selected_atoms`
+    void set_selected_atoms(torch::optional<TorchLabels> selected_atoms);
+
+    /// Serialize a `ModelEvaluationOptions` to a JSON string.
+    std::string to_json() const;
+    /// Load a serialized `ModelEvaluationOptions` from a JSON string.
+    static ModelEvaluationOptions from_json(std::string_view json);
+
+private:
+    torch::optional<TorchLabels> selected_atoms_ = torch::nullopt;
+};
+
+
+/// Metadata about a specific exported model
+class METATENSOR_TORCH_EXPORT ModelMetadataHolder: public torch::CustomClassHolder {
+public:
+    ModelMetadataHolder() = default;
+
+    /// Initialize `ModelMetadata` with the given information
+    ModelMetadataHolder(
+        std::string name_,
+        std::string description_,
+        std::vector<std::string> authors_,
+        torch::Dict<std::string, std::vector<std::string>> references_
+    ):
+        name(std::move(name_)),
+        description(std::move(description_)),
+        authors(std::move(authors_)),
+        references(references_)
+    {
+        this->validate();
+    }
+
+    ~ModelMetadataHolder() override = default;
+
+    /// Name of this model
+    std::string name;
+
+    /// Description of this model
+    std::string description;
+
+    /// List of authors for this model
+    std::vector<std::string> authors;
+
+    /// References for this model. The top level dict can have three keys:
+    ///
+    /// - "implementation": for reference to software used in the implementation
+    ///   of the model
+    /// - "architecture": for reference that introduced the general architecture
+    ///   used by this model
+    /// - "model": for reference specific to this exact model
```

Comment:
For me this is clear but maybe we give and for example?

```suggestion
    /// - "implementation": for reference to software and libraries used in the implementation
    ///   of the model i.e. pytorch: https://dl.acm.org/doi/10.5555/3454287.3455008
    /// - "architecture": for reference that introduced the general architecture
    ///   used by this model
    /// - "model": for reference specific to this exact model.
```